arangod/IResearch/IResearchMetricStats.h

```diff
@@ -0,0 +1,115 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Valery Mironov
+////////////////////////////////////////////////////////////////////////////////
+#pragma once
+
+#include "Cluster/ClusterFeature.h"
+#include "Cluster/ClusterInfo.h"
+#include "IResearch/IResearchDataStore.h"
+#include "Metrics/Guard.h"
+#include "RestServer/arangod.h"
+#include "ApplicationFeatures/ApplicationServer.h"
+
+namespace arangodb::iresearch {
+
+class MetricStats : public metrics::Guard<IResearchDataStore::Stats> {
+  static constexpr std::string_view kShard = ",shard=\"";
+
+ public:
+  static constexpr size_t kSize = 5;
+  static constexpr std::array<std::string_view, kSize> kName = {
+      "arangodb_search_num_docs",       //
+      "arangodb_search_num_live_docs",  //
+      "arangodb_search_num_segments",   //
+      "arangodb_search_num_files",      //
+      "arangodb_search_index_size",     //
+  };
+
+  // toVPack
+  static bool skip(ArangodServer& server, std::string_view labels) {
+    auto const start = labels.find(kShard);
+    TRI_ASSERT(start != std::string_view::npos);
+    std::string /*TODO(MBkkt) Fix cluster info interface*/ shardId{
+        labels.substr(start + 8, labels.size() - (start + 8) - 1)};
```

Comment:
The +8 here look really magic. Is that `kShard.size()`? If so, we should write it as such, to improve understandability.